### ISSUE TITLE
Icon/indicator that user is choosing rpm/flatpak version of an application

### ISSUE THREAD

**Issue by mkrajnak** [flatpak/flatpak#4014]
flatpak-1.8.3-1.fc33.x86_64
xdg-desktop-portal-1.8.0-1.fc33.x86_64
## Description of the problem

Is there a plan to have e.g. a small flatpak (box) icon, or any other indication which would describe to user which icon belongs to a flatpak version or an rpm version of an application ? Currently, there is no way to differentiate between them and I find it confusing: 

### Search overview:

![image](https://user-images.githubusercontent.com/10319675/101366566-629abf00-38a5-11eb-936d-b813ef946875.png)

 ### OpenURI portal:

![image](https://user-images.githubusercontent.com/10319675/101366682-865e0500-38a5-11eb-99a0-f70f7bbea96c.png)

**Comment by alexlarsson** [comment]
I understand that this can be a problem in some cases, but the point of flatpak is to be something in the background, not a technology that a regular user should have to think about, so giving it such a prominent place in the UI like an icon for every app isn't really a good idea.

**Comment by alexlarsson** [comment]
I guess the desktop itself could chose to add an icon in the case where there are both types being displayed at the same time. Thats nothing flatpak can affect though.

**Comment by mkrajnak** [comment]
> I guess the desktop itself could chose to add an icon in the case where there are both types being displayed at the same time. Thats nothing flatpak can affect though.

Right, so if I understand correctly,  should I report this under both gnome-shell and xdg-desktop-portal and wait what their maintainers think about it ? I have minimum knowledge about the implementation but was kinda thinking, is there a way how to do this through desktop files so we would have it fixed on one place instead of two ?

**Comment by saivert** [comment]
I have just resorted to editing the .desktop files for the rpm versions for now. You can make a local override by naming it the same and putting it on ~/.local/share/applications

A script to automate this:

    #!/bin/sh
    sed -e 's/Name=\(.*\)/Name=\1 (rpm)/g' $1 > $HOME/.local/share/applications/$(basename $1)

**Comment by mkrajnak** [comment]
> I have just resorted to editing the .desktop files for the rpm versions for now. You can make a local override by naming it the same and putting it in `~/.local/share/applications`
> 
> A script to automate this:
> 
> ```
> #!/bin/sh
> sed -e 's/Name=\(.*\)/Name=\1 (rpm)/g' $1 > $HOME/.local/share/applications/$(basename $1)
> ```
> 
> Run it with:
> 
> ```
> renameapp.sh /usr/share/applications/firefox.desktop
> ```

 Renaming the applications via script is nice , thanks for that for sure, however when I filled this issue I was looking for more "out of the box" solution. I agree that less-tech-savy users do not care about what is happening behind the scenes (flatpak vs rpm) but then we should not confuse them with 2 icons.

**Comment by chrisawi** [comment]
Nowadays, both versions of Firefox use the same desktop file name (at least on Fedora), so there would only be one icon. This is also true for a large amount of software that embraces the app ID concept.

Marking Flatpak apps with some sort of badge is something that the desktop shell would be able to do, if desired.
